woocommerce/wc-admin
# Remove dynamic content hash from chunk names.

This PR seeks to introduce stability to the filenames used for async code chunks generated by Webpack.

We were previously using `[chunkhash]` in the `chunkFilename` parameter which changes based on the content of the chunk file. This caused a stressful workflow when deploying the plugin to WP.org where the SVN diffs were large and required removing all of the old chunk files so the new ones could be added.

This PR:
- Removes the dynamic `[chunkhash]` from the chunk filename
- Tells the `SplitChunksPlugin` not to generate chunk names - resulting in either numeric IDs, or named files (specified by the `webpackChunkName` magic comment on dynamic `import()`s)
  - Examples: `1.min.js`, `activity-panels-orders.min.js`
- Introduces a new Webpack plugin `AsyncChunkSrcVersionParameterPlugin` that adds the script version parameter to the chunk URLs for cache busting

### Detailed test instructions:

- Remove the `/dist` directory
- Build using `npm run build`
- Verify the filenames in `dist/chunks` do not contain hashes
- Smoke test the WC Admin app functionality
- Verify that the `?ver=` param is set on all JS and CSS chunk requests (using dev tools)



### Changelog Note:



Dev: Make code chunk filenames more stable.
